MudCharts V8Changes

ingkor opened the pull request:
MudCharts Changes (WIP)

Resolves #9910 

I implemented the 3 comments of @henon and @ScarletKuro of issue #9910 & #9907

- Removed MudBlazor.Charts namespace
- Files with multiple classes in split up
- Interpolate methods and classes made internal



## Type of Changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation (fix or improvement to the website or code docs)


## Checklist
<!-- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] The PR is submitted to the correct branch (`dev`).
- [x] My code follows the code style of this project.
- [x] I've added relevant tests.

ScarletKuro committed:
Introduce KeyInterceptorService withs sync/async support to replace KeyInterceptor  (#9896)

henon committed:
MudDialog: Add OnKeyDown and OnKeyUp events (#9906)

ralvarezing committed:
MudDateRangePicker: Incorrect margins and align when on a stack or grid (#9890)

jperson2000 committed:
MudMask: Added XML Documentation for Public Members (#9849)

Co-authored-by: Meinrad Recheis <meinrad.recheis@gmail.com>

danielchalmers committed:
Update KeyInterceptorService API usage (#9912)

igotinfected committed:
MudChip: Improve keyboard accessibility (#9455)

ScarletKuro committed:
MudDataGrid: Fix NRE when property has a nested null value (#9916)

ingkor committed:
removed Mudblazor.Chars namespace

ingkor committed:
removed some extra charts references

ingkor committed:
making interpolate classes internal, removing multiple classes from 1 file

ingkor committed:
some additional Mudblazor.charts removed (tests, old api links)

ingkor committed:
dotnet format

ingkor committed:
additional references deleted

ingkor committed:
donut

henon commented:
> I implemented the 3 comments of @henon and @ScarletKuro of issue #9918 & #9907

#9918 is referencing this PR, which issue did you want to reference?

ingkor commented:
Conflicts may be because I changed all the files to file-scoped namespaces. Let me double check.

I meant to reference this issue #9910 
>For some reason the components is in it's own namespace MudBlazor.Charts, while all of the rest components are in the root MudBlazor. I think there should be consistency that all components should be in MudBlazor because it's the the only namespace we suggest to add to imports.
From brief look it seems like the implementations in the Interpolation folder can be internal, and this classes have public fields:
[MudBlazor/src/MudBlazor/Components/Chart/Interpolation/Matrix.cs](https://github.com/MudBlazor/MudBlazor/blob/4569c85a1dea12f0e7d7a878ef3c4c83c8585b7c/src/MudBlazor/Components/Chart/Interpolation/Matrix.cs#L5-L14)


>instead of public properties.
Multiples classes in one file (not nested), we do not use such practice anymore.

ScarletKuro commented:
> Conflicts may be because I changed all the files to file-scoped namespaces.

I think I would revert that change.
The reason is that it messes up the blame completely which is the fist layer that we use to find a regression.
As far as I know we decided to apply the file-scoped namespaces only if:
1. New file
2. Small amount of code (old file).
3. The file has not much commits done by different people, mainly one person.

ingkor committed:
revert file-scoped namespace

ingkor commented:
So I reverted the File-scoped namespace changes, the git blame will work now.

The merge conflicts is because I changed all the namespaces from `MudBlazor.xxx` to `MudBlazor`

henon commented:
> The merge conflicts is because I changed all the namespaces from `MudBlazor.xxx` to `MudBlazor`

Normally a change on top of the upstream branch (in this case dev) doesn't cause a merge conflict. If your branch is out of date and someone else changed the same files in the upstream branch then you'll get conflicts. Are you sure your branch's starting point was/is up-to-date?

ingkor commented:
>Edit: Merge conflicts are because my branch is behind the MudBlazor branch. Because I have a PR on my dev open, I can't fix it yet. <= Sorry, will do this when the other PR is merged.

This is 50 % the case I think, yes. 
But look at the conflicts, it's all the namespace https://github.com/MudBlazor/MudBlazor/pull/9918/conflicts

henon commented:
With some git tricks you can do it even before. Like check out the **remote dev** into a new local branch (i.e. dev2) then merge that new local branch into this PR's branch and resolve the conflicts.

ingkor committed:
Merge remote-tracking branch 'origin' into MudCharts

ingkor commented:
I'm gonna abandon this one. Merge conflicts were too tricky. Gonna start a new branch and new PR
